src/org/nfrac/comportex/cells.cljc

```diff
@@ -370,16 +371,23 @@
   the proximal excitation but is given keyed by cell rather than by
   column. Normally only cells with some proximal input are included,
   but if `spontaneous-activation?` is true, this is relaxed
-  (i.e. prediction alone could cause activation)."
+  (i.e. prediction alone could cause activation).
+
+  * col-exc is keyed by column as [col 0].
+  * tp-exc is keyed by cell as [col ci]."
   [col-exc tp-exc distal-exc distal-weight spontaneous-activation? depth]
-  (let [;; add TP columns to list of columns to consider (may have no proximal)
-        basic-col-exc (merge-with + col-exc
-                                  (into {} (map (fn [[col _]] [col 0])) (keys tp-exc)))
+  (let [has-tp? (seq tp-exc)
+        ;; add TP columns to list of columns to consider (may have no proximal)
+        basic-col-exc (if has-tp?
+                        (merge-with + col-exc
+                                    (into {} (map (fn [[col _]] [[col 0] 0]))
+                                          (keys tp-exc)))
```

Comment:
(You can ignore the part where I changed your `into` and transducer to a `zipmap`. Your way might be faster since it avoids lazy seqs.)